### PR TITLE
HIVE-22042: Set hive.exec.dynamic.partition.mode=nonstrict by default

### PR DESCRIPTION
**Pull request by jfsii**

### CODE REVIEW

File: ql/src/test/results/clientpositive/load_dyn_part11.q.out [hunk 1]
```diff
@@ -65,2012 +65,2012 @@ POSTHOOK: Input: default@nzhang_part_n0
...
-213	val_213	2010-03-03	11
-146	val_146	2010-03-03	11
-406	val_406	2010-03-03	11
-429	val_429	2010-03-03	11
-374	val_374	2010-03-03	11
+97	val_97	2010-03-03	11
+200	val_200	2010-03-03	11
+400	val_400	2010-03-03	11
+403	val_403	2010-03-03	11
+169	val_169	2010-03-03	11
```
Comment (vineetgarg02) [inline ql/src/test/results/clientpositive/load_dyn_part11.q.out]: I am assuming the results are same but in different order now?

File: ql/src/test/queries/clientpositive/correlationoptimizer13.q [hunk 2]
```diff
@@ -1,5 +1,7 @@
 --! qt:dataset:src
 set hive.mapred.mode=nonstrict;
+set hive.exec.dynamic.partition.mode=strict;
```
[the remark this answers is not in the document]
Comment (jfsii) [reply ql/src/test/queries/clientpositive/correlationoptimizer13.q]: it failed with a NullPointerException without it. I inquired about it with @jcamachor he suggested I should ignore the failure and make it strict, due to the correlationoptimizer not being use or it being deprecated.

File: ql/src/test/results/clientpositive/load_dyn_part11.q.out (hunk 1 above)
Comment (jfsii) [reply ql/src/test/results/clientpositive/load_dyn_part11.q.out]: I manually checked and that is the conclusion I arrived at.

**Comment by vineetgarg02** [comment]
Looks Good.

**Pull request closed without merging**